### ISSUE TITLE
ADAP and NIST

### ISSUE THREAD

**Issue by mbell145** [mzmine/mzmine2#748]
### Basic information

* My operating system: Windows 10
* My MZmine version: 2.51

### What happened

We are trying to build an annotation workflow for GCMS data. We can process the data in mzmine, but we would like to export an annotated peak list table. If we process the data without the Spectral Deconvolution and GC Aligner using MS Search will generate a peak list with the annotations but they have not been spectrally deconvoluted so every peak on an EI spectra gets an annotation. If you take the annotated list and run it through the Spectral Decon we found that the annotations were lost and the running the MS Search on the Spectral Decon did not work with the MS Search. So we follows the ADAP manual and exported an msp file to the NIST MS Search and we can open it and view the matches but MS Search does not have a built in export function to export an annotated peak list. So my question is: Is there a way in MzMine to produce an aligned Spectrally Deconvoluted GCMS (Electron Impact) peak list with NIST annotations?

**Comment by asmirn1** [comment]
**Regarding the workflow using Spectral Deconvolution, GC Aligner, and NIST MS Search...**

You can try using [MSPepSearch](https://chemdata.nist.gov/dokuwiki/doku.php?id=peptidew:mspepsearch) instead of NIST MS Search. I believe, MSPepSearch uses the same matching algorithms and libraries as NIST MS Search and can export annotation results into a csv file.

So your workflow would be:

1. ...
2. spectral deconvolution,
3. alignment,
4. export the produced feature list into a csv file and export the spectra into a msp file,
5. run MSPepSearch to annotate the spectra and save results into a csv file,
6. combine the two produced csv files.

**Comment by mbell145** [comment]
So we tried that, but:

1)      Couldn’t get the most recent version of MSPepSearch to work (would open and immediately close the program… tried both 32 and 64 bit versions)

2)      Got the older 2013 version to work, looked promising but when I submitted, no output file was generated. Maybe I’m wrong but it seems like MSPepSearch is only for MS/MS data, not full scan EI MS data. Either way I tried searching against the NIST mainlib but it doesn’t seem to be working. 

Any other suggestions or troubleshooting ideas?

**Comment by asmirn1** [comment]
Here is an example of the console code to run MSPepSearch with Mainlib and Replib libraries with Quick Identity Search Type.  
```
C:\MSPepSearch_x64\MSPepSearch.exe Q /MinInt 50 /MinMF 800 /HITS 5 /MAIN C:\NISTDEMO\MSSEARCH\MAINLIB /REPL C:\NISTDEMO\MSSEARCH\replib /INP spectra.msp /OUT annotation.txt
```
You can replace the last option /OUT to /OUTTAB to get the annotation results in the tab-separated format. Run MSPepSearch.exe without arguments to see all available options.

I haven't had any problems with running both versions of MSPepSearch in console. I don't have much experience with its GUI though.

If you provide more details on how you run MSPepSearch, I might be able to help more.

Alex

**Comment by mbell145** [comment]
So I have tried

```
C:\2013_06_04_MSPepSearch_x64\MSPepSearch.exe Q /MinInt 50 /MinMF 800 /HITS 5 /MAIN C:\NISTDEMO\MSSEARCH\MAINLIB /INP HarleySpectra.msp /OUT annotation.txt
Initializing libraries 09:43:35...
Libraries initialization complete. 09:43:35...
Error: Cannot open input file "HarleySpectra.msp"
```

but I get the error above. I can import this msp file into NIST MS Search so I do not think it is corrupt. Any suggestions?

**Comment by asmirn1** [comment]
It looks like MSPepSearch cannot find file HarleySpectra.msp. Try using the absolute path to this file.

**Comment by mbell145** [comment]
That worked! But where does the annotation.txt file output too?

I ran this:

```
C:\2013_06_04_MSPepSearch_x64\MSPepSearch.exe Q /MinInt 50 /MinMF 800 /HITS 5 /MAIN C:\NISTDEMO\MSSEARCH\MAINLIB /INP C:\HarleySpectra.msp /OUT annotation.txt
Initializing libraries 10:10:50...
Libraries initialization complete. 10:10:50...
start: 10:10:50
1050 spectra in 4.460 sec, process time = 4.470 sec (read 0.065s print 0.027s init pio 0.000s aux 0.002s sp 0.000s, search only: 0.438s+3.929s=4.366s, 240.48 spec/s) 235.41 spec/s Compared 494743 spec, No spectra compared 0, peaks/spec 34.07
end: 10:10:54
```
But I have searched my C drive and I can't find the file. Was it generated?

**Comment by asmirn1** [comment]
Great! Probably, annotation.txt was saved in the local directory where you are running this command from. You can also specify the absolute path for file annotations.txt (e.g. `C:\annotation.txt`), so it'll be easier to find it.

**Comment by mbell145** [comment]
Awesome! Thank you very much for your help. 

As an additional question, when you called in the second library above

`/REPL C:\NISTDEMO\MSSEARCH\replib`

Can you add different libraries (i.e Golm), and what format do the libraries have to be in to work with MSPepSearch?

**Comment by asmirn1** [comment]
A library has to be readable by NIST MS Search. Actually, MS Search has a tool to convert an MSP file (and some other formats) into the library used by MS Search. It's called `lib2nist.exe` and it's located in the MS Search folder.

**Comment by asmirn1** [comment]
@du-lab @tomas-pluskal Shall we add a section about MSPepSearch into ADAP User Manual, or it is out of scope of MZmine?

**Comment by mbell145** [comment]
Hi, 

I ran into another problem if you have a moment. I was trying to run the newer version of MS PepSEARCH on another computer with the full mainlib and replib (came with software from GCMS vendor), and I had an error message again. 

`C:\2019_02_22_MSPepSearch_x64\MSPepSearch64.exe Q /MinInt 50 /MinMF 800 /HITS 5 /MAIN "C:\Users\Thermo\Program Files (x86)\NIST\MSSEARCH\mainlib" /REPL "C:\Users\Thermo\Program Files (x86)\NIST\MSSEARCH\replib" /INP HarveyADAPspectra.msp /OUTTAB annotation.txt
Initializing libraries 12:48:25...
MS Libraries initiation error -10`

Do you have experience with this error?

**Comment by asmirn1** [comment]
No, I haven't seen this error before. Could you check if these libraries work with MS Search?

**Comment by mbell145** [comment]
They do work within the thermo file. I also copied both libraries out to another computer and into the NISTDEMO, and they worked fine there as long as I used the 2019 version of the MSPepSearch. The 2013 version threw access denied errors on the copied libraries. We would prefer to use it on the same computer as the instrument however, which is the one that was throwing the Initiation error.

**Comment by asmirn1** [comment]
It's possible that the latest libraries won't work with the old version of MSPepSearch. Can you use the 2019 version of MSPepSearch on the instrument computer?

**Comment by mbell145** [comment]
I was using the latest version of the MSPepSearch when I ran into the Initiation error

**Comment by asmirn1** [comment]
Maybe, the path to the files isn't correct. This path `C:\Users\Thermo\Program Files (x86)\NIST\MSSEARCH\mainlib` looks a bit strange because of `Program Files (x86)` in the middle, but I might be wrong.

If the path to both libraries is correct, then I don't know what might be the reason for the initialization error.

**Comment by tomas-pluskal** [comment]
@asmirn1 I think mentioning MSPepSearch in the ADAP manual is definitely a good idea!

It would be even better to connect the ADAP deconvolution modules with the Local spectral database search module developed by @robinschmid so the users can search spectral databases (including NIST?) directly from MZmine.

**Comment by asmirn1** [comment]
@tomas-pluskal @robinschmid I was planning to look into Local spectral database search module for quite a while, but I've been busy with other projects. As I understand, we would have to separate the spectrum building part of Local spectral database search into a separate module. This way, we could use all other algorithms for spectrum construction. I'm still planning to look into it, maybe sometime this winter.

**Comment by robinschmid** [comment]
Within the next three weeks, we are going to introduce some structural changes to cover the relationship of signals, i.e., different ions of the same molecule. The ADAP correlation results (clustered spectra) could then fit into this structure. 

And we should definitely include the ADAP results into the spectral library matching. 
See #728 for a collection of thoughts on the spectral library matching module.

**Comment by Sukis123** [comment]
@asmirn1 Using MSPepSearch for searching against NIST libraries is a great idea! But I am not sure if it is possible to use retention index information for scoring?  For example, there is an option to choose the type of column for retention index in NIST MS Search. It is also possible to use the retention index for scoring in MS-DIAL.

**Comment by asmirn1** [comment]
Right now MZmine doesn't calculate retention indices, but I think @robinschmid is currently working on it. After this work is done, we should be able to use retention indices with NIST MS Search.

**Comment by Sukis123** [comment]
@asmirn1 Understand, many thanks.
